src/monster.h

```diff
@@ -164,6 +165,9 @@ class monster : public Creature, public JsonSerializer, public JsonDeserializer
         int calc_movecost( const tripoint &f, const tripoint &t ) const;
         int calc_climb_cost( const tripoint &f, const tripoint &t ) const;
 
+        bool is_dangerous_field( const field &fld ) const;
+        bool is_dangerous_field( const field_entry &entry ) const;
```

Comment:
Character doesn't share any inheritance with monster. I could generalize a Creature version of the call if we want, but it would literally just say ```return entry.is_dangerous``` and the Character/monster versions would end with ```return Creature::is_dangerous_field(fld)```.

Inheritance structure
* Creature -> Character -> Player -> NPC (for now, ideally NPC should inherit directly from Character)
* Creature -> Monster